Add fp8 content for hipify

PoodleWang opened the pull request:
Summary:
Add fp8 hipify content.

Test plan:
Internal test for NV and AMD GPUs. 

Internal usage for meta. [D67305195]

PoodleWang commented:
@pytorchbot label "topic: not user facing"

PoodleWang commented:
@albanD

jeffdaily reviewed (commented):
If you look at the amd_hip_fp8.h header in ROCm 6.3, you'll see that both ocp and fnuz types are supported.  gfx942 uses fnuz, gfx1200 uses ocp.  We will need to do something more creative than this in order to support different gfx targets.

jithunnair-amd commented on torch/utils/hipify/cuda_to_hip_mappings.py:
```diff
@@ -576,6 +576,7 @@
         ("library_types.h", ("hip/library_types.h", CONV_INCLUDE, API_RUNTIME)),
         ("cuComplex.h", ("hip/hip_complex.h", CONV_INCLUDE, API_RUNTIME)),
         ("cuda_fp16.h", ("hip/hip_fp16.h", CONV_INCLUDE, API_RUNTIME)),
+        ("cuda_fp8.h", ("hip/hip_fp8.h", CONV_INCLUDE, API_RUNTIME)),
```
@jeffdaily So should this map to amd_hip_fp8.h instead?

PoodleWang commented:
> If you look at the amd_hip_fp8.h header in ROCm 6.3, you'll see that both ocp and fnuz types are supported. gfx942 uses fnuz, gfx1200 uses ocp. We will need to do something more creative than this in order to support different gfx targets.

I made a new change.  ("__NV_E4M3", ("__HIP_E4M3", CONV_OTHER, API_RUNTIME)), then in the /torch/utils/hipify/hipify_python.py. It will transfer the __HIP_E4M3 to __HIP_E4M3_FNUZ or __HIP_E4M3_OCP

PoodleWang commented:
@jeffdaily would you take a look at this pull request?

Commit:
Hipify Nvidia GPU benchmark for AMD GPUs (#144351)

Summary:

Hipify Nvidia GPU benchmark for AMD GPUs

Since CUBLASLT_MATMUL_DESC_FAST_ACCUM is not able to be transferred by the AMD hipify commdn. Let's skip gpu monitor for AMD.

Test Plan:
a100
command:
 buck2 run @//mode/opt //ai_codesign/nonprod/ehsanardestani/training/bench/gemm_mem:gemm_microbenc
output:
1024 2048 4096 0 0 100 fp8 0 5 0 / 100

gemm 0.03872 us 443695 Tflops 
mem 52.3101 us bw (GB/s) 320.726
mixed 52.1005 us 0.03872 GEMM us 443695 0.000743179 % GEMM time 
memory bound kernel: 10.585 us, 6340.02 GB/s,

mi300x
buck2 run @//mode/opt-amd-gpu //ai_codesign/nonprod/ehsanardestani/training/bench/gemm_mem:gemm_microbench

1024 2048 4096 111 111 100 fp8 0 5 0 / 100

gemm 0.04886 us 351614 Tflops 
mem 20.4482 us bw (GB/s) 820.473
mixed 16.677 us 0.04886 GEMM us 351614 0.00292978 % GEMM time 
memory bound kernel: 31.5776 us, 2125.2 GB/s,

Differential Revision: D67305195
